collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -73,9 +68,8 @@ public void doScrape() throws Exception {
                 mBeanNames.removeAll(beanConn.queryNames(name, null));
             }
 
-            for (ObjectName name : mBeanNames) {
-                scrapeBean(beanConn, name);
-            }
+            mBeanNames.parallelStream().forEach(name -> scrapeBean(beanConn, name));
```

Comment:
parallelStream was only added in 1.8, it'll be at least 2020 before 1.8 is so widespread that we can safely assume that everyone uses at least 1.8.

In addition, what's the cpu impact of doing this in parallel? Spiky cpu usage due to monitoring is undesirable.